sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -622,10 +659,12 @@ def process_bundle_progress_metadata_request(self,
                                                request,  # type: beam_fn_api_pb2.ProcessBundleProgressMetadataRequest
                                                instruction_id  # type: str
                                               ):
+    # type: (...) -> beam_fn_api_pb2.InstructionResponse
     return beam_fn_api_pb2.InstructionResponse(
         instruction_id=instruction_id,
-        process_bundle_progress=beam_fn_api_pb2.
+        process_bundle_progress_metadata=beam_fn_api_pb2.
         ProcessBundleProgressMetadataResponse(
+            # FIXME: incompatible type "List[MonitoringInfo]"; expected "Optional[Mapping[str, Any]]"
             monitoring_info=SHORT_ID_CACHE.getInfos(
                 request.monitoring_info_id)))
```

Comment:
Note this change and the FIXME.  I'm not sure what to do about this. 

I got complaints that `InstructionResponse(process_bundle_progress=...)` was receiving the wrong type.  I'm not sure whether this is the right resolution.

Also, `ProcessBundleProgressMetadataResponse(monitoring_info=...)`is also receiving the wrong type.  It's supposed to be a mapping not a list.  Should I change `SHORT_ID_CACHE.getInfos` to return a dictionary? 

Do you know if this code is being hit?  If so, any theories on how it's working?

Reply:
This code is not being used currently and it should be a mapping.

Yes, please update getInfos to return that mapping.